### PR TITLE
Enhance Multilingual Support with Chinese and German Label Recognition

### PR DESCRIPTION
**Pull request by neozhu**
**Description:**

This Pull Request introduces significant enhancements to our application's language recognition capabilities. With the integration of new features, we're now extending our support to include both Chinese and German languages, catering to a broader user base and improving our application's versatility in multilingual environments.

**Key Features and Changes:**

1. **Chinese Label Recognition with PaddleOCR**: We've implemented Chinese label recognition utilizing the powerful PaddleOCR framework. This addition enables our application to accurately recognize and process labels in Chinese, facilitating users in Chinese-speaking regions to interact with our application more effectively.

2. **German and Chinese Language Support**: Building on our commitment to support multiple languages, we've added recognition capabilities for German alongside Chinese. This ensures that users from German-speaking countries can also benefit from our application's features without language barriers.

**Benefits:**

- **Broader Reach**: By supporting more languages, our application becomes accessible to a wider audience, enhancing user experience and satisfaction.
- **Improved Accuracy**: The use of PaddleOCR for Chinese label recognition promises high accuracy and efficiency, ensuring that our application meets the needs of users requiring Chinese language support.
- **Enhanced User Experience**: Multilingual support significantly improves the overall user experience, allowing users to interact with the application in their preferred language.

image

### CODE REVIEW

File: label_studio_ml/examples/paddleocr/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+label-studio-ml
```
Comment (niklub) [inline label_studio_ml/examples/paddleocr/requirements.txt:1]: it is better to pinpoint versions in requirements, to avoid failed dependencies in future docker image rebuilds

File: label_studio_ml/examples/tesseract/docker-compose.yml [hunk 2]
```diff
@@ -12,15 +12,15 @@ services:
...
   minio:
     container_name: minio
-    image: quay.io/minio/minio
+    image: bitnami/minio:latest
     env_file:
       - example.env
     ports:
-      - 9000:9000
-      - 9001:9001
+      - 9003:9000
```
Comment (niklub) [inline label_studio_ml/examples/tesseract/docker-compose.yml:22]: is there any reason why did you change the defaults ports? if not - I'd suggest to keep them as before avoiding unnecessary changes

File: label_studio_ml/examples/tesseract/example.env [hunk 3]
```diff
@@ -1,11 +1,11 @@
-LABEL_STUDIO_HOST=http://host.docker.internal:8080
-LABEL_STUDIO_ACCESS_TOKEN=<access token if using local file storage>
+LABEL_STUDIO_HOST=http://10.33.1.37:8089
```
Comment (niklub) [inline label_studio_ml/examples/tesseract/example.env:1]: please rollback to initial default parameters

**Review by niklub — CHANGES_REQUESTED** [review]
Thanks for the great contribution! A couple of comments left to address, and then we can merge it to the master

**Comment by neozhu** [comment]
> Thanks for the great contribution! A couple of comments left to address, and then we can merge it to the master

Thank you for the review and the feedback. I appreciate your suggestions and will make the necessary changes right away.

File: label_studio_ml/examples/tesseract/example.env (hunk 3 above)
Comment (neozhu) [reply label_studio_ml/examples/tesseract/example.env:1]: reverted

File: label_studio_ml/examples/paddleocr/paddleocr_ch.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            result = context.get('result')[0]
+            meta = self._extract_meta({**task, **result})
+            x = meta["x"]*meta["original_width"]/100
+            y = meta["y"]*meta["original_height"]/100
+            w = meta["width"]*meta["original_width"]/100
+            h = meta["height"]*meta["original_height"]/100
+
+            crop_img = IMG.crop((x, y, x + w, y + h))
+
+            # 使用 PaddleOCR 进行文字识别
```
[the remark this answers is not in the document]
Comment (neozhu) [reply label_studio_ml/examples/paddleocr/paddleocr_ch.py]: done

File: label_studio_ml/examples/paddleocr/paddleocr_ch.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        task = tasks[0]
+        img_path_url = task["data"]["ocr"]
+
+
+        context = kwargs.get('context')
+        if context:
+            if not context["result"]:
+                return []
+
+            IMG = self.load_image(img_path_url)
```
[the remark this answers is not in the document]
Comment (neozhu) [reply label_studio_ml/examples/paddleocr/paddleocr_ch.py:71]: done

File: label_studio_ml/examples/paddleocr/requirements.txt (hunk 1 above)
Comment (neozhu) [reply label_studio_ml/examples/paddleocr/requirements.txt:1]: done

**Comment by neozhu** [comment]
I've made the necessary changes and have committed it again. Could you please review it once more?

**Comment by neozhu** [comment]
I'm closing this PR and have already resubmitted two PRs, #466  and #465  . Please review them when you have time. Thank you.
